hub/integrations/pytorch/pytorch.py

```diff
@@ -256,14 +256,16 @@ def _generate_shared_memory_names(self, chunk_names: Set[str]):
     def _numpy_from_chunk(self, index: int, key: str, chunk):
         """Takes a list of chunks and returns a numpy array from it"""
         chunk_engine = self.all_chunk_engines[key]
-        value = chunk_engine.read_sample_from_chunk(index, chunk)
+        value = chunk_engine.read_sample_from_chunk(index, chunk, cast=False)
 
         # typecast if incompatible with pytorch
-        if value.dtype == "uint16":
-            value = value.astype("int32")
-        elif value.dtype == "uint32" or value.dtype == "uint64":
-            value = value.astype("int64")
-        return torch.tensor(value)  # type: ignore
+        tensor_dtype = chunk_engine.tensor_meta.dtype
+        casts = {"uint16": "int32", "uint32": "int64", "uint64": "int64"}
+        to_dtype = casts.get(value.dtype.name)
+        if to_dtype:
+            value = value.astype(to_dtype)
```

Comment:
why do we need two casts? a single cast in torch.as_tensor would be better.
For eg. if tensor_meta dtype is uint32 and the dtype of value is uint16, currently you would first cast to int32 using astype and then subsequently cast to int64 in .as_tensor. We should have just a single cast for this in .as_tensor and remove the as_type logic altogether.